knowage-vue/src/modules/documentExecution/dashboard/widget/SelectorWidget/SelectorWidget.vue

```diff
@@ -0,0 +1,451 @@
+        },
+        dateRangeSelectionChanged() {
+            if (this.editorMode) return
+            const tempDateValues = [] as string[]
+            for (let i = 0; i < this.initialOptions.rows.length; i++) {
+                const iniitalOption = this.initialOptions.rows[i].column_1
+                const tempDate = moment(iniitalOption, dashboardDescriptor.selectionsDateMultiFormat).valueOf()
+                if ((!this.startDate || tempDate >= this.startDate.getTime()) && (!this.endDate || tempDate <= this.endDate.getTime())) tempDateValues.push(iniitalOption)
+            }
+            const tempSelection = this.createNewSelection(tempDateValues)
+            updateStoreSelections(tempSelection, this.activeSelections, this.dashboardId, this.setSelections, this.$http)
+        },
+        updateActiveSelectionsWithMultivalueSelection(tempSelection: ISelection) {
+            const index = this.activeSelections.findIndex((activeSelection: ISelection) => activeSelection.datasetId === tempSelection.datasetId && activeSelection.columnName === tempSelection.columnName)
+            if (index !== -1) {
+                this.activeSelections[index] = tempSelection
+            } else {
+                this.activeSelections.push(tempSelection)
+            }
+        },
+        createNewSelection(value: (string | number)[]) {
+            return { datasetId: this.propWidget.dataset as number, datasetLabel: this.getDatasetLabel(this.propWidget.dataset as number), columnName: this.propWidget.columns[0]?.columnName ?? '', value: value, aggregated: false, timestamp: new Date().getTime() }
+        },
+        getDatasetLabel(datasetId: number) {
+            const index = this.datasets.findIndex((dataset: IDataset) => dataset.id.dsId == datasetId)
+            return index !== -1 ? this.datasets[index].label : ''
+        },
+        onSelectionsDeleted(selections: any) {
+            const index = selections.findIndex((selection: ISelection) => selection.datasetId === this.propWidget.dataset && selection.columnName === this.propWidget.columns[0]?.columnName)
+            if (index !== -1) this.removeDeafultValues()
+        }
+    }
+})
+</script>
+
+<style lang="scss">
+.selector-widget {
+    overflow-y: auto;
+    .multi-select {
+        display: flex;
+        align-items: center;
+        .multi-select-label {
+            text-overflow: ellipsis;
+            overflow: hidden;
+            white-space: nowrap;
+        }
+    }
+    .vertical-layout {
+        display: flex;
+        flex-direction: column;
+    }
+    .horizontal-layout {
+        display: flex;
+    }
+    .grid-layout {
+        display: flex;
+        flex-direction: row;
+        flex-wrap: wrap;
+        overflow-y: auto;
+    }
+}
+.selectorCustomDropdownPanel {
+    color: unset;
+}
+</style>
+<style lang="scss" scoped>
```

Comment:
This scrollbar style is used often. Should we put it in a common __dashboard.scss_ file?

Reply:
Fixed